lib/route_format.rb

```diff
@@ -2,7 +2,7 @@
 
 module RouteFormat
   def self.username
-    /[%\w.\-]+?/
+    /[%\w.\-#{SiteSetting.allowed_unicode_username_characters}]+?/
```

Comment:
Since we can configure the allowed characters to include "/", should we add a test to verify that Rails handles an encoded forward slash as we expect?